### PR TITLE
Add codegen for MLIR enums

### PR DESCRIPTION
**Pull request by segeljakt**
Credit for most of these changes goes to Frej!

All types in MLIR are structural. Therefore this commit adds a
conversion from nominal enums into structural enums.

The Arc-Script code:

```
    enum Foo {
        Bar(i32),
        Baz(f32),
    }
```

Becomes the MLIR type:
```
    !arc.enum<crate_Foo_Bar : i32 , crate_Foo_Baz : f32>
```

### CODE REVIEW

File: arc-script/arc-script-test/compile/src/snapshots/ast@structs.arc.snap [hunk 1]
```diff
@@ -4,8 +4,12 @@ expression: s
 input_file: arc-script-test/compile/src/tests/expect_pass/structs.arc
 
 ---
+fun foo(a: { c: i32, b: i32 }, b: { b: i32, c: i32 }) -> { a: { c: i32, b: i32 }, xyz: i32, d: { b: i32, c: i32 } } {
```
Comment (frej) [inline arc-script/arc-script-test/compile/src/snapshots/ast@structs.arc.snap:7]: Why does a patch implementing enums change golden output for structs? Is this something that should go in a separate patch?

**Review by frej — APPROVED** [review]
LGTM although it looks like there are unrelated changes.

File: arc-script/arc-script-test/compile/src/snapshots/ast@structs.arc.snap (hunk 1 above)
Comment (segeljakt) [reply arc-script/arc-script-test/compile/src/snapshots/ast@structs.arc.snap:7]: strange, I split this into a separate commit now

**Comment by frej** [comment]
Looks good, merge away!
